sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -767,7 +767,9 @@ def _resolve_arm_id_or_upload_dependencies(self, component: Component) -> None:
             self._all_operations, self._operation_scope, self._operation_config
         ).get_asset_arm_id
 
-        self._resolve_dependencies_for_component(component, resolver)
+        self._resolve_dependencies_for_component(
+            component, resolver, resolve_jobs_inputs=isinstance(component, PipelineComponent)
```

Comment:
comment why specifal for PipelineComponent

Reply:
hmm... simple answer is that only pipeline component has jobs : ) done some refactor so it looks easier to understand.